### PR TITLE
Pd 144 add list of metrics

### PR DESCRIPTION
**Pull request by andreipope**

### CODE REVIEW

**Comment by andreipope** [comment]
> All our metrics are gauges - did you find information that states otherwise?

@prashantsr  Hey, I was just thinking that counters are used for values that always increment, while gauges are used for values that can either increment or decrement. So, I assumed that some metrics like the total bytes read from a volume or the time spent doing I/Os are actually counters. Anyway, if I'm wrong, it'll take just 2 minutes to update the table.

**Comment by prashantsr** [comment]
In our cases, all of these are gauges (we may have to fix a few and make them counters) but today they are not. I'd go with what we have at the moment.

**Comment by andreipope** [comment]
> In our cases, all of these are gauges (we may have to fix a few and make them counters) but today they are not. I'd go with what we have at the moment.

Got it. Now it's updated. All metrics are gauges. Thanks!

**Comment by andreipope** [comment]
@prashanthpx @harsh-px  All requests have been taken care of. Could we merge this pull request?

**Comment by andreipope** [comment]
@prashantsr @harsh-px All requests have been taken care of. Could we merge this pull request?

**Comment by prashantsr** [comment]
I can approve this now but I know Paul is making some changes to metric names and this doc will need to be updated.

**Comment by nathan-px** [comment]
@andreipope, From my conversation with @pault84, the updated list of metrics is here: https://portworx.slack.com/files/U08HPV1JQ/FMZ5MHR8S/metrics

* A metric with "deprecated" prepended indicates that the link was changed
* Ones marked with "deprecated: not supported" should be removed
* The metric subsystems aren't shown in this list, but are still required (volume, node_stats, etc)

**Comment by pault84** [comment]
Related to PWX-9891

**Comment by andreipope** [comment]
> @andreipope, From my conversation with @pault84, the updated list of metrics is here: https://portworx.slack.com/files/U08HPV1JQ/FMZ5MHR8S/metrics

Thanks, @nathan-px! Looks I can't access the link though
Screenshot 2019-09-14 20 21 43
.

File: content/shared/metrics/portworx-metrics.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+| px_cluster_disk_utilized_bytes | Gauge | Used storage in the Portworx cluster | |
+| px_cluster_disk_total_bytes | Gauge | Total storage in the Portworx cluster | |
+| px_cluster_pendingio | Gauge | Total bytes (read/write) being currently processed | |
+| px_cluster_status_cluster_size | Gauge | Total cluster size | |
+| px_cluster_status_nodes_offline | Gauge | Number of offline nodes | |
+| px_cluster_status_nodes_online | Gauge | Number of online nodes | |
+| px_cluster_status_nodes_storage_down | Gauge | Number of storage down nodes | |
+| px_cluster_status_cluster_quorum | Gauge| Cluster quorum | 1 = in quorum, 0 = not in quorum |
+| px_cluster_status_storage_nodes_decommissioned | Gauge | Number of decommissioned storage nodes | |
+| px_cluster_status_storage_nodes_online | Gauge | Number of nodes proving storage which are online | These nodes participate in quorum |
```
Comment (nathan-px) [inline content/shared/metrics/portworx-metrics.md]: proving or providing?

File: content/shared/metrics/portworx-metrics.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+## Cluster stats
+
+| Name | Description | Notes |
+| :--- | :--- | :--- |
+| size | The total number of nodes in your Portworx cluster. | |
+| quorum | Indicates if the cluster is in quorum. | 1 = in quorum, 0 = not in quorum |
+| nodes_online | The number of online nodes in your cluster | It includes storage and storageless nodes |
+| nodes_offline | The number of offline nodes. | It includes storage and storageless nodes. |
+| nodes_storage_down | The number of storage nodes where the storage is full or can't be accessed. <!-- Not sure I got this right. Original description: "Number of storage nodes where the storage that is full or down"--> | |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr Please see this comment.

File: content/shared/metrics/portworx-metrics.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+| storage_nodes_online | The number of online storage nodes. | These nodes participate in quorum. <!-- Is this correct> --> |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+| storage_nodes_offline | The number of offline storage nodes. <!-- Is this correct? Original description: "Number of nodes with storage that are marked offline". --> | These nodes participate in quorum. <!-- Is this correct? --> |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+| storage_nodes_decommissioned | The number of decommisisioned storage nodes. | |
+| cpu_percent | The average CPU usage for the nodes within your Portworx cluster. | |
+| memory_utilized_percent | The average memory utilization for the nodes within your Portworx cluster. | |
+| disk_total_bytes | The total amount of storage space for this node <!-- Is this correct? If so, it should be probably placed under node stats --> | Expressed in bytes. |
+| disk_available_bytes | The amount of available storage space for this node. <!-- Is this correct? If so, it should be probably placed under node stats --> | Expressed in bytes. |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+| disk_utilized_bytes | The amount of used space for this node.  <!-- Is this correct? If so, it should be probably placed under node stats --> | Expressed in bytes. |
+| pendingio | The number of read and write operations currently in progress for this node  <!-- Is this correct? If so, it should be probably placed under node stats --> | |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+| read_latency_seconds | The average amount of time spent per read operation from this disk. | Expressed in seconds. |
+| write_latency_seconds | The average amount of time spent per write operation to this disk. | Expressed in secnds. |
+| num_reads | The total number of successfull read operations for this disk. | |
+| num_writes | The total number of successfull write operations for this disk. | |
+
+## Pool Stats
+
+| Name | Description | Notes |
+| :--- | :--- | :--- |
+| written_bytes | The total amount of bytes written to this pool since the last interval. <!-- I don't really get this. How long is the interval? -->| |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+| total_bytes | The total amount of spac for this pool | Expressed in bytes. |
+
+
+## Volume stats
+
+
+| Name | Description | Notes |
+| :--- | :--- | :--- |
+| usage_bytes | The amount of used storage space for this volume. | |
+| capacity_bytes | The configured size in bytes for this volume. <!-- I don't get this. Maybe we should provide more details --> | |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+| write_hits | The number of write hits. | |
+| write_miss | The number of write misses. | |
+| block_size | The block size. | |
+| mode | Cache mode <!-- Probably needs more details. Maybe list the nodes.--> | |
+
+## Network stats
+
+| Name | Description | Notes |
+| :--- | :--- | :--- |
+| bytessent | The number of bytes sent during this interval. <!-- For me, this is a bit unclear. Can we define the interval?--> | |
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+| Name | Description | Notes |
+| :--- | :--- | :--- |
+| node_status | The status of this node. | Please consult the [API reference](https://libopenstorage.github.io/w/master.generated-api.html#status) for more details. | |
+| free_mem | The amount of free memory. | Expressed in bytes. |
+| total_mem | The total amount of memory. | Expressed in bytes. |
+| used_mem | The amount of used memory. | Expressed in bytes. |
+| cpu_usage | The percentage of CPU usage. | |
+
+
+## Proc stats <!-- Proc filesystem stats? -->
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

File: content/shared/metrics/portworx-metrics.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+
+#### Relevant metrics
+
+1. disk_utilized_bytes, disk_available_bytes
+2. cpu_percent
+3. size, storage_nodes_online
+4. size, quorum
+
+
+<!-- Do we want to include the list of deprecated metrics?
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @nathan-px @prashantsr Please see this comment.

File: content/shared/metrics/portworx-metrics.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+2. Use the latency `heatmap` to identify the volumes with high read/write latency.
+3. Find the volume name from all volume latency panel.
+4. Use the volume name to filter all metrics for that volume.
+
+#### Relevant metrics
+
+1. Disk stats: write_seconds, read_seconds
+2. Disk stats: read_latency_seconds
+3. Disk stats: write_latency_seconds
+4. Check other volume metrics, as you see fit. <!-- Shouldn't this be somthing like: "Check other disk metrics, as you see fit." -->
```
Comment (andreipope) [inline content/shared/metrics/portworx-metrics.md]: @prashantsr

**Comment by nathan-px** [comment]
@prashanthpx @pault84 Is this still on hold, or is it going out with 2.2.1?

**Comment by nathan-px** [comment]
> @prashanthpx @pault84 Is this still on hold, or is it going out with 2.2.1?

@pault84 @prashanthpx bumping this for 2.3.2

**Comment by nathan-px** [comment]
Last I heard, these metrics were not yet live, is that still the case @pault84 @prashanthpx ?

**Comment by pault84** [comment]
Correct. Master branch will go into 2.4.1 i think.

**Comment by nathan-px** [comment]
This is being automated, see #714

**Pull request closed without merging**